### PR TITLE
Add a multidoc preview link to the side navigation. 

### PR DESCRIPTION
**Pull request by yebai**
See e.g. https://turing.ml/multidoc-preview/AbstractMCMC/dev/

### CODE REVIEW

**Review by cpfiffer — APPROVED** [review]

File: _data/toc.yml [hunk 1]
```diff
@@ -92,14 +92,7 @@
...
 - title: "API"
-  url: "docs/library"
-  children:
-    - title: "Turing"
-      url: "docs/library"
-    - title: "AdvancedHMC"
-      url: "docs/library/advancedhmc/"
-    - title: "Bijectors"
-      url: "docs/library/bijectors/"
+  url: "https://turing.ml/multidoc-preview/"
```
Comment (devmotion) [inline _data/toc.yml:95]: Could we use a more "official" location? Maybe
```suggestion
  url: "https://turing.ml/api/"
```
or
```suggestion
  url: "https://turing.ml/docs/library/"
```
?

File: multidoc/make.jl [hunk 2]
```diff
@@ -56,8 +80,9 @@ MultiDocumenter.make(
 
 # Deploy to Github with running as a Github action
 if haskey(ENV, "GITHUB_ACTIONS")
-    isdir("multidoc-preview") || mkdir("multidoc-preview")
-    multidocroot = normpath(joinpath(@__DIR__, "../multidoc-preview/"))
+    output_dir = "library"
+    isdir(output_dir) || mkdir(output_dir)
```
Comment (devmotion) [inline multidoc/make.jl:84]: This can be simplified to
```suggestion
    mkpath(output_dir)
```

File: multidoc/make.jl [hunk 3]
```diff
@@ -56,8 +80,9 @@ MultiDocumenter.make(
...
+    multidocroot = normpath(joinpath(@__DIR__, "../", output_dir))
```
Comment (devmotion) [inline multidoc/make.jl:85]: Should be sufficient:
```suggestion
    multidocroot = normpath(joinpath(@__DIR__, "..", output_dir))
```

Comment (yebai) [reply multidoc/make.jl:85]: I fixed these issue in [96abc8c](https://github.com/TuringLang/turing.ml/commit/96abc8c744782c6c566080128370b0f7a6c425e3)

**Pull request merged**